linux/keyman-config/keyman_config/gnome_keyboards_util.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/python3
+import logging
+import os
+from gi.repository import Gio
+
+from gi.overrides.GLib import Variant
+
+
+class GnomeKeyboardsUtil():
+    def __init__(self):
+        self.input_sources = Gio.Settings.new("org.gnome.desktop.input-sources")
+
+    def read_input_sources(self):
+        sourcesVal = self.input_sources.get_value("sources")
+        sources = self._convert_variant_to_array(sourcesVal)
+        return sources
+
+    def write_input_sources(self, sources):
+        sourcesVal = self._convert_array_to_variant(sources)
+        self.input_sources.set_value("sources", sourcesVal)
+
+    def _convert_variant_to_array(self, variant):
+        if variant is None:
+            return []
+
+        values = []
+        # a(ss)
+        nChildren = variant.n_children()
+        for i in range(nChildren):
+            # (ss)
+            val = variant.get_child_value(i)
+            typeVariant = val.get_child_value(0)
+            type = typeVariant.get_string()
+            idVariant = val.get_child_value(1)
+            id = idVariant.get_string()
+            values.append((type, id))
+        return values
+
+    def _convert_array_to_variant(self, array):
+        if len(array) == 0:
+            return None
+
+        children = []
+        for (type, id) in array:
+            typeVariant = Variant.new_string(type)
+            idVariant = Variant.new_string(id)
+            child = Variant.new_tuple(typeVariant, idVariant)
+            children.append(child)
+        return Variant.new_array(None, children)
+
+
+__is_gnome_shell = None
+
+
+def is_gnome_shell():
+    global __is_gnome_shell
+
+    if __is_gnome_shell is None:
+        code = os.system('pidof gnome-shell >/dev/null 2>&1')
+        __is_gnome_shell = (code == 0)
+    return __is_gnome_shell
+
+
+def _reset_gnome_shell():
+    # used in unit tests
+    global __is_gnome_shell
+
+    __is_gnome_shell = None
+
+
+def get_keyboard_id(keyboard, packageDir, ignore_language=False):
```

Comment:
Can we use a different name for this function? Rationale: `keyboard_id` is already a fairly heavily used term in Keyman (and equates to the basename of the keyboard file sans extension (e.g. `khmer_angkor`).

Reply:
done